pkg/workflow/providers/multicluster/deploy.go

```diff
@@ -96,13 +101,22 @@ func selectPolicies(policies []v1beta1.AppPolicy, policyNames []string) ([]v1bet
 	return selectedPolicies, nil
 }
 
-func loadComponents(ctx context.Context, cli client.Client, af *appfile.Appfile, components []common.ApplicationComponent) ([]common.ApplicationComponent, error) {
+func loadComponents(ctx context.Context, renderer oamProvider.WorkloadRenderer, cli client.Client, af *appfile.Appfile, components []common.ApplicationComponent, ignoreTerraformComponent bool) ([]common.ApplicationComponent, error) {
 	var loadedComponents []common.ApplicationComponent
 	for _, comp := range components {
 		loadedComp, err := af.LoadDynamicComponent(ctx, cli, comp.DeepCopy())
 		if err != nil {
 			return nil, err
 		}
+		if ignoreTerraformComponent {
+			wl, err := renderer(comp)
```

Comment:
can we use comp.Type to judge here? so we don't need to render, this is a complex calculate which affect the performance

Reply:
The component type is diverse and irregular